vmngclient/session.py

```diff
@@ -462,18 +469,13 @@ def __init__(
         self.subdomain = subdomain
         super().__init__(url, port, username, password, timeout)
 
-    def login(self) -> None:
-        """Logs in to vManage API as Provider using username/password and switches to Tenant."""
-        super().login()
-        self.__switch_to_tenant()
-
-    def __get_tenant_id(self) -> str:
+    def __get_tenant_id(self, subdomain: str) -> str:
         """Gets tenant UUID for tenant subdomain.
         Returns:
             tenant UUID
         """
         tenants = self.get_data('/dataservice/tenant')
-        tenant_ids = [tenant.get('tenantId') for tenant in tenants if tenant['subDomain'] == self.subdomain]
+        tenant_ids = [tenant.get('tenantId') for tenant in tenants if tenant['subDomain'] == subdomain]
         assert len(tenant_ids) > 0, f"Tenant not found for subdomain: {self.subdomain}"
```

Comment:
self.subdomain is mixed with subdomain
I would keep using self.subdomain from constructor so advanced user can specify it when creating object.
I would keep private methods parametrized with subdomain - however make this parameter optional so we fallback to constructor default when not provided and overwrite self.subdomain when parameter provided
__switch_to_tenant, __get_tenant_id

Reply:
done, if subdomain is provided `self.subdomain = subdomain`